crates/core/src/processor.rs

```diff
@@ -0,0 +1,158 @@
+//! Processor is reponsible for running the module.
+//! This run is sound execution of the module.
+//! This will be most abstract layer of the processor.
+
+use alloy_dyn_abi::DynSolValue;
+use alloy_merkle_tree::standard_binary_tree::StandardMerkleTree;
+use alloy_primitives::{hex::FromHex, FixedBytes, Keccak256, B256, U256};
+use anyhow::Result;
+use hdp_primitives::processed_types::{cairo_format::AsCairoFormat, v1_query::ProcessedResult};
+use serde::Serialize;
+use std::{path::PathBuf, str::FromStr};
+
+use hdp_provider::evm::{AbstractProvider, AbstractProviderConfig};
+
+use crate::cairo_runner::run::Runner;
+
+pub struct Processor {
+    runner: Runner,
+    _provider: AbstractProvider,
+}
+
+#[derive(Debug, Serialize)]
+pub struct ProcessorResult {
```

Comment:
hmm result suggest Result<> maybe ProcessorOutput is better name?

Reply:
Processor input should also be defined here
why processor has provider ideally it should not